### DISCUSSION TITLE
Is it possible to encode large JPEG/PNG images to JPEG?

### DISCUSSION THREAD

**Discussion by Aytsemik** [SixLabors/ImageSharp#2671] [Q&A] [answered]
I'm attempting to encode a 100MB image on macOS and encountering issues. It works for smaller images like those under 10MB. The problem persists on Linux as well. I'm utilizing C# ASP.NET .NET 7.

**Comment by dlemstra** [comment]
What kind of issues are you encountering? And the size of the file does not matter. The dimensions (width x height) of the image do.

**Reply by Aytsemik** [reply]
Thank you for the response. The image dimensions are  1943 × 1296 JPEG , and size is 89 MB. I'm getting an error 
`Image cannot be loaded. Available decoders:\n - PNG : PngDecoder\n - GIF : GifDecoder\n - TIFF : TiffDecoder\n - JPEG : JpegDecoder\n - PBM : PbmDecoder\n - QOI : QoiDecoder\n - Webp : WebpDecoder\n - TGA : TgaDecoder\n - BMP : BmpDecoder\n`

**Reply by Aytsemik** [reply]
I've tried smaller image with dimensions 3197 × 3795 it worked.

**Reply by JimBobSquarePants** [reply]
That means the image does not contain a header that is recognised by any of our decoders. It’s nothing to do with the size. 

Can you use a hex editor and post a screen shot of the leading bytes of the file?

**Reply by tocsoft** [reply, marked as answer]
sounds like that particular image your using isn't a real/standard JPEG (as we can understand it), that error your seeing is our format detection failing to determine the decoder to use.

I would suggest you first verify it really is a jpeg and not some other format that just happens to have a file with a jpeg file extension.

**Reply by Aytsemik** [reply]
I used  an online tool, is this ok ?

<img width="1166" alt="Screenshot 2024-02-21 at 4 20 34 PM" src="https://github.com/SixLabors/ImageSharp/assets/5806803/71587cda-d0b4-4f19-a732-349cfc946106">

**Reply by tocsoft** [reply]
that's a PSD/photoshop file not a jpeg.

**Reply by Aytsemik** [reply]
Yes, you are right. I tried more than 50 images and didn't think it could be related to that. All were from the same source, and the only difference was the size. Thanks for the response
